pkgs/top-level/all-packages.nix

```diff
@@ -14594,6 +14594,13 @@ in
     boost = boost159;
   };
 
+  mysql80 = callPackage ../servers/sql/mysql/8.0.x.nix {
+    inherit (darwin) cctools developer_cmds;
+    inherit (darwin.apple_sdk.frameworks) CoreServices;
+    boost = boost169;
+    stdenv = pkgs.clangStdenv;
```

Comment:
Please use the default boost if mysql compiles with it, and build with the default stdenv.

Reply:
Re: `boost`: mysql 8.0.16 seemed to be a bit particular about this. The [release notes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-16.html) specify the use of boost 1.6.9. With either default `boost` (aka `boost167`) or newer `boost170`, the build fails with similar messages. 

```
-- BOOST_INCLUDE_DIR /nix/store/h6rp8r64zdndyy4hfpd3zbdav4bpqs2x-boost-1.67.0-dev/include
-- LOCAL_BOOST_DIR
-- LOCAL_BOOST_ZIP
-- Could not find (the correct version of) boost.
-- MySQL currently requires boost_1_69_0

CMake Error at cmake/boost.cmake:103 (MESSAGE):
  You can download it with -DDOWNLOAD_BOOST=1 -DWITH_BOOST=<directory>

  This CMake script will look for boost in <directory>.  If it is not there,
  it will download and unpack it (in that directory) for you.

  If you are inside a firewall, you may need to use an https proxy:

  export https_proxy=http://example.com:80

Call Stack (most recent call first):
  cmake/boost.cmake:291 (COULD_NOT_FIND_BOOST)
  CMakeLists.txt:836 (INCLUDE)
```

(I'm pretty sure this error was the same regardless of host platform and of other build options.)

Reply:
Re: `stdenv` - This was one of the hardest things for me to work out (in no small part b/c I'm inexperienced with this - but I think there's also something wonky going on.)

If one simply changes to use the default `stdenv`, then the build still works on Darwin, but it fails on Linux with:

```
[  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvector.cpp.o
[  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvectr32.cpp.o
[  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvectr64.cpp.o
[  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/wintz.cpp.o
[  7%] Linking CXX static library libicuuc.a
[  7%] Built target icuuc
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/aj50pkqzx2z8z48jbz68f2pwp9xkvnyy-mysql-8.0.16.drv' failed with exit code 2
error: build of '/nix/store/aj50pkqzx2z8z48jbz68f2pwp9xkvnyy-mysql-8.0.16.drv' failed
```

A few data-points/factoids;

- As a general pro-forma thing, I just tried rebasing to use current `nixpkgs@master` (in case stdenv might be diff now), and it still had the same problem.
- When first working on this PR, I wound up trying several flavors of `stdenv` under Darwin+Linux. (`mysql` is a large package, so that took...  patience 😉 ) Setting it to `clangStdenv` was the only thing that worked on both. However, I didn't go in planning/expecting to explore those permutations, so I don't have very systemic notes.
- When first working on the patch, there were a few times where I tried running build commands by hand and used `make -j2` or `make -j3` to speed it up. This manifested new errors - and eventually I realized that those errors didn't happen in a clean build using single-threaded `make`.   Feels like there may be a sequencing issue in the build system.
- The statement `enableParallelBuilding=true` was copied over from `5.7.x.nix`; I didn't register that it might be relevant until @aanderse pointed to its existence.
- I took a shot in the dark and tried permuting `stdenv,enableParallelBuilding` within a Linux system. On `clangStdenv`, the value of `enableParallelBuilding` doesn't seem to have much effect. On the default `stdenv`, it changes the outcome (*although both are failures*). Compare:

    * `Linux, stdenv=clangStdenv, enableParallelBuilding=true`: *Seems to build OK. (Haven't done a new/full run, but the first 10+ min of building seem fine.)*
    * `Linux, stdenv=clangStdenv, enableParallelBuilding=false`: *Seems to build OK. (Haven't done a new/full run, but the first 10+ min of building seem fine.)*
    * `Linux, stdenv=stdenv, enableParallelBuilding=true`: *Fails fairly quickly, with:*

        ```
        [  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvector.cpp.o
        [  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvectr32.cpp.o
        [  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/uvectr64.cpp.o
        [  7%] Building CXX object extra/icu/source/common/CMakeFiles/icuuc.dir/wintz.cpp.o
        [  7%] Linking CXX static library libicuuc.a
        [  7%] Built target icuuc
        make: *** [Makefile:152: all] Error 2
        builder for '/nix/store/aj50pkqzx2z8z48jbz68f2pwp9xkvnyy-mysql-8.0.16.drv' failed with exit code 2
        error: build of '/nix/store/aj50pkqzx2z8z48jbz68f2pwp9xkvnyy-mysql-8.0.16.drv' failed
        ```

    * `Linux, stdenv=stdenv, enableParallelBuilding=false`: Fails fairly quickly, with:

        ```
        > extern "C" struct rpl_transaction_ctx_service_st {
        >   int (*set_transaction_ctx)(
        >       Transaction_termination_ctx transaction_termination_ctx);
        > } * rpl_transaction_ctx_service;
        > int set_transaction_ctx(
        >     Transaction_termination_ctx transaction_termination_ctx);
        > #include <mysql/service_rpl_transaction_write_set.h>
        > struct Transaction_write_set {
        >   unsigned int m_flags;
        >   unsigned long write_set_size;
        >   unsigned long long *write_set;
        > };
        > extern "C" struct transaction_write_set_service_st {
        >   Transaction_write_set *(*get_transaction_write_set)(
        >       unsigned long m_thread_id);
        > } * transaction_write_set_service;
        > Transaction_write_set *get_transaction_write_set(unsigned long m_thread_id);
        > #include <mysql/service_rules_table.h>
        > #include <string>
        CMake Error at /tmp/nix-build-mysql-8.0.16.drv-0/mysql-8.0.16/cmake/do_abi_check.cmake:86 (MESSAGE):
          ABI check found difference between
          /tmp/nix-build-mysql-8.0.16.drv-0/mysql-8.0.16/include/mysql/services.h.pp
          and /tmp/nix-build-mysql-8.0.16.drv-0/mysql-8.0.16/build/abi_check.out


        make[2]: *** [CMakeFiles/abi_check.dir/build.make:57: CMakeFiles/abi_check] Error 1
        make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/abi_check.dir/all] Error 2
        make: *** [Makefile:152: all] Error 2
        builder for '/nix/store/r9gx8an5v5wg5fgja9f7s2mx75zpyhv0-mysql-8.0.16.drv' failed with exit code 2
        error: build of '/nix/store/r9gx8an5v5wg5fgja9f7s2mx75zpyhv0-mysql-8.0.16.drv' failed
        ```

Reply:
Marking resolved based on @orivej's updates.